### PR TITLE
Use a temp file, when writing the cached image

### LINKED ISSUE #3960: When interrupting a download, the cache image gets corrupted
> ❌  Unable to load cached images: loading cached images: loading image /home/anders/.minikube/cache/images/k8s.gcr.io/k8s-dns-kube-dns-amd64_1.14.13: Docker load /tmp/k8s-dns-kube-dns-amd64_1.14.13: command failed: docker load -i /tmp/k8s-dns-kube-dns-amd64_1.14.13
stdout: 
stderr: Error processing tar file(exit status 1): unexpected EOF

It works anyway (downloading the image again), but the output is ugly enough.

### PR DESCRIPTION
**Pull request by afbjorklund**
Closes #3960

### CODE REVIEW

File: pkg/minikube/machine/cache_images.go [hunk 1]
```diff
@@ -314,10 +314,19 @@ func CacheImage(image, dst string) error {
...
 	if err != nil {
 		return err
 	}
-	defer f.Close()
-	return tarball.Write(tag, img, nil, f)
+	err = tarball.Write(tag, img, nil, f)
+	if err != nil {
+		return err
+	}
+	f.Close()
```
Comment (tstromberg) [inline pkg/minikube/machine/cache_images.go]: Please check the error code to close!

File: pkg/minikube/machine/cache_images.go [hunk 2]
```diff
@@ -314,10 +314,19 @@ func CacheImage(image, dst string) error {
 	}
 
 	glog.Infoln("OPENING: ", dstPath)
-	f, err := os.Create(dstPath)
+	tmpPath := dstPath + ".tmp"
+	f, err := os.Create(tmpPath)
```
Comment (tstromberg) [inline pkg/minikube/machine/cache_images.go]: Will this fail if dstPath.tmp exists already? 

I'd personally rather see ioutil.TempFile used here, passing in this same directory as an argument.

Comment (afbjorklund) [reply pkg/minikube/machine/cache_images.go]: It should truncate, but we can use random file names

File: pkg/minikube/machine/cache_images.go (hunk 1 above)
Comment (afbjorklund) [reply pkg/minikube/machine/cache_images.go]: I think we did it with `defer` before... WIll check

**Pull request merged**